### DISCUSSION TITLE
Adding tvOS support to existing RN iOS project

### DISCUSSION THREAD

**Discussion by bohlec** [react-native-tvos/react-native-tvos#416] [Q&A] [answered]
Hi all -

Apologies if this is well-known or covered elsewhere, but I've been unable to find a way to get this working with my existing react native iOS project. Is there a guide to supporting both iOS and tvOS in one project?

I think my root question is related to the swapping of the `react-native` dependency for the corresponding `react-native-tvos` artifact. It seems like to support both platforms, we'd somehow have to install both packages.

I currently have added a tvOS target to my project and have multiple `target` sections in my podfile that is referencing the `react-native-tvos` dependency. My tvOS target builds correctly. However, switching back to my iOS target causes build failures with tvOS-specific code references:

<img width="494" alt="image" src="https://user-images.githubusercontent.com/1735896/208505125-fed5e80b-1917-429e-9df5-c647ee22187e.png">

Does anyone have any suggestions on how to support both iOS and tvOS targets?

**Comment by bohlec** [comment, marked as answer]
Well that error looks like simply a bug fixed here: https://github.com/react-native-tvos/react-native-tvos/pull/357

Looks like I have to move to 0.69!

**Reply by douglowder** [reply]
Correct 👍

**Reply by bohlec** [reply]
@douglowder - thanks for the response. I'm running into issues with `expo-updates`. Do you have a patch for this package somewhere? I've done the usual changes, but am still having build failures.

**Reply by douglowder** [reply]
Supporting Expo is on my roadmap (I work there :) ) and will be a bit complex for tvOS. I don't have a working patch for expo-updates yet.

**Reply by bohlec** [reply]
@douglowder - ooh good to know you have a bit of a vested interest in supporting that. I'd love to be able to use Expo without the requirement of these patches for each desired dependency! Thanks for the replies and all you do.
